include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -366,10 +363,11 @@ struct __sycl_scan_by_segment_impl
     }
 };
 
-template <typename _BackendTag, typename Policy, typename InputIterator1, typename InputIterator2,
-          typename OutputIterator, typename T, typename BinaryPredicate, typename Operator, typename Inclusive>
-OutputIterator
-__scan_by_segment_impl_common(__internal::__hetero_tag<_BackendTag>, Policy&& policy, InputIterator1 first1,
+// TODO required to move into oneapi::dpl::__par_backend_hetero namesace
+template <typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator, typename T,
+          typename BinaryPredicate, typename Operator, typename Inclusive>
+oneapi::dpl::__internal::__enable_if_hetero_execution_policy<Policy, OutputIterator>
+__scan_by_segment_impl_common(oneapi::dpl::__internal::__device_backend_tag, Policy&& policy, InputIterator1 first1,
```

Comment:
This probably should be named `__pattern_hetero_scan_by_segment` or something like that.  
In any case, this reshuffling / naming changes belong in a different PR.  

The important part is to come to an agreement about what is in the `hetero` space, and what is in the `device_backend` space.  I believe this function stays in the `hetero` space.

Reply:
I've dealt with this in #2201